libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -731,3 +777,75 @@ void test_MQTT_Subscribe_Publish_With_Qos_2( void )
                        MQTT_ProcessLoop( &context, MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
     TEST_ASSERT_TRUE( receivedUnsubAck );
 }
+
+void test_MQTT_Connect_LWT( void )
+{
+    int secondTcpSocket;
+    NetworkContext_t secondNetworkContext = { 0 };
+    bool sessionPresent;
+    MQTTContext_t secondContext;
+
+    /* Establish a second TCP connection with the server endpoint, then
+     * a TLS session. The server info and credentials can be reused. */
+    TEST_ASSERT_EQUAL( OPENSSL_SUCCESS, Openssl_Connect( &secondNetworkContext,
+                                                         &serverInfo,
+                                                         &opensslCredentials,
+                                                         TRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                                         TRANSPORT_SEND_RECV_TIMEOUT_MS ) );
+    TEST_ASSERT_NOT_EQUAL( -1, secondNetworkContext.socketDescriptor );
+    TEST_ASSERT_NOT_NULL( secondNetworkContext.pSsl );
+
+    /* Establish MQTT session on top of the TCP+TLS connection. */
+    useLWTClientIdentifier = true;
+    establishMqttSession( &secondContext, &secondNetworkContext, true, &sessionPresent );
+
+    /* Subscribe to LWT Topic. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic(
+                           &context, TEST_MQTT_LWT_TOPIC, MQTTQoS0 ) );
+
+    /* Abruptly terminate TCP connection. */
+    ( void ) Openssl_Disconnect( &secondNetworkContext );
+
+    /* Run the process loop to receive the LWT. Allow some more time for the
+     * server to realize the connection is closed. */
+    TEST_ASSERT_EQUAL( MQTTSuccess,
+                       MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
```

Comment:
As the LWT message is expected to only come at QoS 0 PUBLISH, is it required to use a large receive timeout for the LWT message?

Reply:
The timeout's also to account for the time needed for the server to recognize the connection is closed before it sends the LWT